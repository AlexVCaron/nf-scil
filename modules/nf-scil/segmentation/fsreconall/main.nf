process SEGMENTATION_FSRECONALL {
    tag "$meta.id"
    label 'process_single'

    // Note. Freesurfer is already on Docker. See documentation on
    // https://hub.docker.com/r/freesurfer/freesurfer
    container "freesurfer/freesurfer:7.1.1"

    input:
        tuple val(meta), path(anat)

    output:
        path("*__recon_all")                    , emit: recon_all_out_folder
        path "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    // Note. In dsl1, we used an additional option:   -parallel -openmp $params.nb_threads.
    // Removed here.
    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    export SUBJECTS_DIR=.

    recon-all -i $anat -s ${prefix}__recon_all -all

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        freesurfer: 7.1.1
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    recon-all -help

    mkdir ${prefix}__recon_all

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        freesurfer: 7.1.1
    END_VERSIONS
    """
}
