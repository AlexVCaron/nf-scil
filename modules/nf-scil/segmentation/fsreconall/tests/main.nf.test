// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test segmentation/fsreconall
nextflow_process {

    name "Test Process SEGMENTATION_FSRECONALL"
    script "../main.nf"
    process "SEGMENTATION_FSRECONALL"

    tag "modules"
    tag "modules_nfcore"
    tag "segmentation"
    tag "segmentation/fsreconall"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    config "./nextflow.config"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "freesurfer.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("segmentation - fsreconall") {

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/anat_image.nii.gz"),
                        file("\${test_data_directory}/license.txt")
                    ]}
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.recon_all_out_folder.get(0).get(1)).name
                ).match() }
            )
        }

    }

}
