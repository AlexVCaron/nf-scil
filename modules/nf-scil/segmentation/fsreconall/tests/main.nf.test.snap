{
    "segmentation - fsreconall": {
        "content": [
            [
                
            ],
            [
                "versions.yml:md5,12027dd965b94e4d68b7e8a630a04d00"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-21T12:40:38.69571"
    }
}